src/pyhf/pdf.py

```diff
@@ -222,12 +222,22 @@ def __init__(self, spec, **config_kwargs):
             spec, self.channel_nbins
         )
 
-        poiname = config_kwargs.get('poiname', 'mu')
+        # measurement_name is explicitly removed in cases where the model is created using poi_name, multiple
+        # measurements are there but the first one is taken or if measurement is defined as None in the argument
+        config_kwargs.pop('measurement_name', None)
```

Comment:
these lines could be removed entirely. `_ModelConfig()` should never be called with `measurement_name`. if this is the case, then logic needs to be fixed, but we can file an issue to fix this beyond the PR.

Reply:
Yeah, I would remove the comment. I don't think it is helpful and currently as is seems confusing.

Reply:
I've filed #836 to keep track.

Reply:
Indeed as in the first place, this should already be handled.